packages/web/src/App.tsx

```diff
@@ -1,21 +1,123 @@
-import "bootstrap/dist/css/bootstrap.min.css";
-import "./App.css";
 import React from "react";
-import { Container } from "reactstrap";
+import {
+  createBrowserRouter,
+  RouterProvider,
+  Route,
+  Routes,
+  Navigate,
+  Outlet,
+} from "react-router-dom";
+import { Container, Row, Col, Spinner } from "reactstrap";
 import { useAuth } from "./hooks/useAuth";
+import LeetrbooApp from "./LeetrbooApp";
+import SignIn from "./SignIn";
+import SignUp from "./SignUp";
+import ResetPassword from "./ResetPassword";
+import SetNewPassword from "./SetNewPassword";
+import { AuthRedirect } from "./AuthRedirect";
 
-console.log(import.meta.env.VITE_APP_SUPABASE_URL);
+    <Routes>
+      <Route path="/auth/callback" element={<AuthRedirect />} />;
+      <Route path="/reset-password" element={<ResetPassword />} />
+      <Route path="/set-new-password" element={<SetNewPassword />} />
+      <Route path="/auth/callback" element={<AuthRedirect />} />
+    </Routes>;
+    
+const ProtectedRoute = () => {
+  const { user, loading } = useAuth();
 
-type Props = {};
+  if (loading) {
+    return (
+      <Container
+        className="d-flex justify-content-center align-items-center"
+        style={{ height: "100vh" }}
+      >
+        <Spinner />
+      </Container>
+    );
+  }
 
-const App: React.FC<Props> = ({ }) => {
-  const user = useAuth();
+  if (!user) {
+    return <Navigate to="/signin" replace />;
+  }
+
+  return <Outlet />;
+};
+
+// Auth Layout wrapper
+const AuthLayout = () => {
+  const { user, loading } = useAuth();
+
+  if (loading) {
+    return (
+      <Container
+        className="d-flex justify-content-center align-items-center"
+        style={{ height: "100vh" }}
+      >
+        <Spinner />
+      </Container>
+    );
+  }
+
+  // Modify this check to allow access to set-new-password route even with a session
+  const isPasswordResetRoute = window.location.pathname === "/set-new-password";
+  if (user && !isPasswordResetRoute) {
+    return <Navigate to="/" replace />;
+  }
 
   return (
-    <Container fluid className="p-0">
-{user == null?<div>no user</div>:<div>user</div>}
+    <Container className="mt-5">
+      <Row className="justify-content-center">
+        <Col md={6}>
+          <Outlet />
+        </Col>
+      </Row>
     </Container>
   );
 };
 
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <ProtectedRoute />,
+    children: [
+      {
+        path: "/",
+        element: <LeetrbooApp />,
+      },
+    ],
+  },
+  {
+    path: "/",
+    element: <AuthLayout />,
+    children: [
+      {
+        path: "signin",
+        element: <SignIn />,
+      },
+      {
+        path: "signup",
+        element: <SignUp />,
+      },
+      {
+        path: "reset-password",
+        element: <ResetPassword />,
+      },
+      {
+        path: "set-new-password",
```

Comment:
This seems to confirm that you could have this as part of the protected routes. When Supabase sends you the link to reset the password, it effectively signs you in.